### PR TITLE
Create clean up function for simulation manager

### PR DESCRIPTION
**Pull request by knarfnitram**
Add a simple clean up function for the simulation manager, which simply deletes all folders and files within the main folder (Place where the simulation is performed/written) - Has not been tested so far for the cluster.

### CODE REVIEW

**Comment by knarfnitram** [comment]
@isteinbrecher any ideas why the performacne tests are failing?

**Comment by isteinbrecher** [comment]
@knarfnitram there already exists a `clean_simulation_directory` function in `utility` that basically contains the same code (looks like we found the same stackexhange post). Additionally this existing function asks the user if the directory should be cleaned.

I think the best way to go forward is to add an optional parameter to the existing function `ask_before_clean` to turn on/off if the user should be asked. Also maybe `ensure_clean_simulation_directory` would be a better name. What do you think?

Regarding the performance tests, depending on the overall load on the test machine they might not pass (does not hinder the progress of the pull request). I had a look and everything looks fine.

**Comment by knarfnitram** [comment]
Makes totally sense to reuse the clean functionality, especially if we were inspired by the same stack exchange post. 

Yes, the name sounds better. 
I have renamed and updated the simulation manager and also added the option `ask_before_clean`.

Regarding the performance checks, I thought that this might be the reason, but thanks for clearifying that!

File: tests/testing_simulation_manager.py [hunk 1]
```diff
@@ -256,6 +256,24 @@ def test_simulation_manager_batch_file(self):
             additional_identifier=2,
         )
 
+    def test_simulation_manager_cleaned_start_directory(self):
```
Comment (isteinbrecher) [inline tests/testing_simulation_manager.py:259]: Please add a doc string shortly describing what this test tests.

File: tests/testing_simulation_manager.py [hunk 2]
```diff
@@ -256,6 +256,24 @@ def test_simulation_manager_batch_file(self):
...
+        manager = self.create_simulations(convergence_base_dir)
+
+        # Ensure clean state before adding next simulation
+        manager.ensure_clean_simulation_directory()
+
+        # check if the folder still exists
+        self.assertTrue(os.path.isdir(manager.path))
+
+        # check if folder is empty
+        self.assertFalse(len(os.listdir(manager.path)))
```
Comment (isteinbrecher) [inline tests/testing_simulation_manager.py:275]:
```suggestion
        self.assertFalse(len(os.listdir(manager.path))==0)
```

File: meshpy/utility.py [hunk 3]
```diff
@@ -143,17 +143,27 @@ def get_min_max_coordinates(nodes):
...
+    ----
+    sim_dir:
+        Path to a directory
+    ask_before_clean: bool
+        Flag which indicates whether the user must confirm removal of files and directories
     """
 
     # Check if simulation directory exists.
     if os.path.exists(sim_dir):
-        print(f'Path "{sim_dir}" already exists')
```
Comment (isteinbrecher) [inline meshpy/utility.py:154]: Shouldn’t this be the other way? If ask_before_clean is True we want to show the message and ask the user. Now it is the other way around.

**Comment by isteinbrecher** [comment]
@knarfnitram what is the state of this pull-request?

File: meshpy/utility.py (hunk 3 above)
Comment (knarfnitram) [reply meshpy/utility.py:154]: yes, you are right, i have inverted the flag.

**Review by isteinbrecher — APPROVED** [review]

**Comment by isteinbrecher** [comment]
@knarfnitram looks good to me, feel free to merge

**Comment by knarfnitram** [comment]
So unfortunally it says:
You’re not [authorized](https://docs.github.com/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches) to merge this pull request.

**Pull request merged**